### ISSUE TITLE
How to run griddedInitialization

### ISSUE THREAD

**Issue by chiarabruni** [firelab/windninja#359]
Hi!

I want to use a  weather forecasts grid and for initialization on WindNinja.
I think I have to use the griddedInitialization  option in the CLI, I have the Input_speed_grid and the Input_dir_grid and the dtm of my study area. 
I keep getting "Exception caught: multiple occurrences" after the run, and I can't understand where my error is.

Thanks a lot

**Comment by nwagenbrenner** [comment]
Yes, you could use the griddedInitialization option in the CLI. Note that this option isn't yet described in our documentation and hasn't been tested much. It's mostly been used for internal purposes. It should work for you though. To use this option, you'll have to supply your gridded wind speed and direction files in AAIGrid format. Make sure you don't have any other initialization methods specified in your configuration file. You can see #269 for more details.

**Comment by chiarabruni** [comment]
I managed to download your trial data from #269 and they worked for me.  Thank you Natalie

**Comment by chiarabruni** [comment]
Anotehr question: the name of the output I get is :"name of the elevation_file" + "mesh resolution" + "mesh unit".
I am working with a big amount of gridded data from different days and hours. Is there a way to automatically re-name the outputs in the .cfg file, according to the date/hour of the gridded forecast ?

**Comment by nwagenbrenner** [comment]
@chiarabruni sorry for the slow reply. You can set the output directory with `output_path` in the cfg, but there is not currently a way to set the filenames for the outputs. You'd have to do that as a post-processing step.

**Comment by chiarabruni** [comment]
Thank you Natalie for your help. Is it posible to run griddedInitialization without the cfg file, isn't it? 
I mean, writing all the options directly in the cli. I am keeping getting some exceptions.

**Comment by nwagenbrenner** [comment]
Yeah, you should be able to supply all of the CLI arguments on the command line, something like:

`WindNinja_cli --num_threads=4 --vegetation=trees ...etc`

If you can't get that to work send me the command your using and error your getting and I'll take a look.

**Comment by chiarabruni** [comment]
Thank you Natalie,
with the .cfg file I had no problems.
I attach the command I am using (ascii output version).
I keep getting "Exception caught: unknown option -t".


[cabreira_16_04_2020.pdf](https://github.com/firelab/windninja/files/4486780/cabreira_16_04_2020.pdf)

**Comment by nwagenbrenner** [comment]
I see a few things:

Make sure all of the dashes are double dashes `--`. It was hard to tell for sure copying from the pdf, but when I pasted your command it looked like some of the dashes were only single dashes `-`.

`units_ascii_resolution` should be `units_ascii_out_resolution`.

`units_output_speed_grid` should be `output_speed_units`.

`units_input_speed_grid` should be `input_speed_units`.

**Comment by chiarabruni** [comment]
I changed the commands you suggested, and checked the dashes, thanks.
It changed the error I get as well,  I am now getting "Exception caught: unknown option -û"
I attach the commands in .txt

[
[script_16_04_2020.txt](https://github.com/firelab/windninja/files/4488092/script_16_04_2020.txt)
](url)

**Comment by nwagenbrenner** [comment]
I see some strange characters in front of the first four arguments. It looks like one of each of those dashes is not an actual dash, but some other character. Please double check those double dashes and try again. Let me know if that fixes your problem.

**Comment by ncubukcu** [comment]
Is there a way to add time dimension to riddedInitialization input .asc files? As I see in the sample set, if we are running multiple time steps, we need to run each steps separately. Thanks

**Comment by nwagenbrenner** [comment]
Right now there's not a way to run multiple time steps with the gridded initialization. You would need to run a separate simulation for each time step. We could look into adding this functionality in the future though.

**Comment by ncubukcu** [comment]
griddedInitialization with mass&momentum solver runs fine with fixed vegetation but if you change elevation file to an lcp file, it hangs at 
Run 0: Reading elevation file...
Any suggestion? This is valid for CLI version, not sure about the GUI.  
Thanks

**Comment by ncubukcu** [comment]
Actually I just realized that  windninja hangs even with mass solver. It hangs at further stage. See here
Run 0: Reading elevation file...
Run 0: Run number 0 started with 4 threads.
Run 0: Generating mesh...
Run 0: Initializing flow...
The entire test run I use is
[here](https://drive.google.com/file/d/1Ki7m3A0moyEE-jQVEojiJiXFUI05ecSK/view?usp=sharing)

It seems to me that both (mass or momentum) have the same trouble and that might be the projection. So it  is not the solver that is issue here but it is some arrangement of gridded input files that is not acceptable. Do I need to supply the same projection? If that is the case then why do we have .prj for each .asc files? Obviously I am all over and no idea where to start but I will try to unify the input projections. Hope it will help. Thanks

**Comment by ncubukcu** [comment]
No luck. Even making the projection same as input lcp file did not help. WindNinja starts running and then freezes as described above.  My version is 
WindNinja version: 3.6.0
SCM version: 0dbb3f3519feb84d930fc06db43170f80bbd2ba7
Release date: 2020/07/07
OpenMP enabled (64)

Thanks

**Comment by nwagenbrenner** [comment]
@ncubukcu I'm not sure griddedInitialization has been tested with an lcp before since it was added as kind of an internal option. Now that it's being more generally used, we should sort out these issues and ensure it's fully functional. I need permission to access your input data at the link you provided. Once I get your data I'll take a closer look and hopefully get this issue resolved.

**Comment by ncubukcu** [comment]
Hi Natalie
Please go ahead and take a look at my test case. I updated the link today with input projection. Thanks

**Comment by nwagenbrenner** [comment]
What is the test.cfg in the test case? It is not a configuration file, it looks like a DEM, but it is partially corrupted or something.

**Comment by ncubukcu** [comment]
Interesting. Somehow that did not make it to the tart file. Here is the cfg. Apologies for that. 
#Cfg File for W123N038
num_threads=2
elevation_file=Input/W123N038.lcp
initialization_method=griddedInitialization
input_speed_grid=Input/W123N038-20191130:00_vel.asc
input_dir_grid=Input/W123N038-20191130:00_ang.asc
input_speed_units=mps
input_wind_height=10
units_input_wind_height=m
time_zone=GMT
momentum_flag=true
number_of_iterations=100
output_wind_height=20.0
output_speed_units=mph
units_output_wind_height=ft
output_path=./
mesh_resolution=250.0
units_mesh_resolution=m
write_goog_output=true
goog_out_resolution=500
units_goog_out_resolution=m
write_shapefile_output=false
write_ascii_output=true
write_farsite_atm=true

**Comment by nwagenbrenner** [comment]
It looks like your velocity and angle grids don't cover your DEM. The extent of the LCP is

-2340060, -2180580
1944000, 2181100

and the extent of the velocity grid is:

-2616657, -2397954
1717640, 1936342

The input wind grids have to completely cover the DEM.

![Screenshot from 2020-10-07 12-51-16](https://user-images.githubusercontent.com/3696653/95374992-4033f780-089c-11eb-9c62-ae3432868f31.png)

**Comment by ncubukcu** [comment]
Ouch.. I was trying so many things out of desperation and I accidently created (-) cell size. The cell size should be positive. I actually saw in another thread that model grid should cover the entire dem so I made it larger than DEM. But I did not see the error I made in the cell size. Thank you so much Natalie. It is running now..  I guess my real issue was the model domain. It won't work even if DEM and model domains are exactly the same. Model domain must be larger in each direction I guess. Thanks
